packages/bot-web-ui/src/stores/run-panel-store.js

```diff
@@ -4,23 +4,82 @@ import { error_types, unrecoverable_errors, observer, message_types } from '@der
 import { contract_stages } from '../constants/contract-stage';
 import { switch_account_notification } from '../utils/bot-notifications';
 
+const statistics_storage_key = 'statistics_cache';
+
+const empty_statistic = {
+    lost_contracts: 0,
+    number_of_runs: 0,
+    total_profit: 0,
+    total_payout: 0,
+    total_stake: 0,
+    won_contracts: 0,
+};
+let get_statistics;
+
 export default class RunPanelStore {
     constructor(root_store) {
         this.root_store = root_store;
         this.dbot = this.root_store.dbot;
         this.registerCoreReactions();
+
+        get_statistics = JSON.parse(sessionStorage.getItem(statistics_storage_key));
+
+        reaction(
+            () => this.statistics,
+            statistics => {
+                const { client } = this.root_store.core;
+                const stored_statistics = JSON.parse(sessionStorage.getItem(statistics_storage_key)) ?? {};
+
+                const new_statistics = { statistic: statistics };
+                stored_statistics[client.loginid] = new_statistics;
+
+                sessionStorage.setItem(statistics_storage_key, JSON.stringify(stored_statistics));
+            }
+        );
+
+        reaction(
+            () => this.root_store.core.client.loginid,
+            () => {
+                const { core, transactions, journal, contract_card } = this.root_store;
+                const { client } = core;
+
+                if (client.is_logged_in) {
+                    const statictics = JSON.parse(sessionStorage.getItem(statistics_storage_key));
+                    const journal_messages = JSON.parse(sessionStorage.getItem('journal_cache'));
+                    const transaction_elements = JSON.parse(sessionStorage.getItem('transaction_cache'));
+
+                    this.statistics =
+                        statictics && statictics[client.loginid]
+                            ? statictics[client.loginid].statistic
+                            : empty_statistic;
+
+                    journal.unfiltered_messages =
+                        journal_messages && journal_messages[client.loginid]
+                            ? journal_messages[client.loginid].journal_message
+                            : [];
```

Comment:
- Property `journal_message` should be plural `journal_messages`, we're not retrieving a single journal message.
- Could be shortened as below:

```suggestion
                    journal.unfiltered_messages = journal_messages?.[client.loginid]?.journal_message
```